pkg/resource/deploy/step.go

```diff
@@ -942,32 +932,48 @@ func (s *ImportStep) Apply(preview bool) (resource.Status, StepCompleteFunc, err
 		}
 
 		if len(inputProperties) == 0 {
-			logging.V(9).Infof("Importing %v with required properties", s.URN())
-
-			for _, p := range r.InputProperties {
-				if p.IsRequired() {
-					k := resource.PropertyKey(p.Name)
-					s.new.Inputs[k] = s.old.Inputs[k]
-				}
-			}
+			logging.V(9).Infof("Importing %v with all properties", s.URN())
+			s.new.Inputs = s.old.Inputs.Copy()
 		} else {
 			logging.V(9).Infof("Importing %v with supplied properties: %v", s.URN(), inputProperties)
-
 			for _, p := range inputProperties {
 				k := resource.PropertyKey(p)
 				if value, has := s.old.Inputs[k]; has {
 					s.new.Inputs[k] = value
 				}
 			}
 		}
-	} else {
-		// Set inputs back to their old values (if any) for any "ignored" properties
-		processedInputs, res := processIgnoreChanges(s.new.Inputs, s.old.Inputs, s.ignoreChanges)
-		if res != nil {
-			return resource.StatusOK, nil, res.Error()
+
+		// Check the provider inputs for consistency. If the inputs fail validation, the import will still succeed, but
+		// we will display the validation failures and a message informing the user that the failures are almost
+		// definitely a provider bug.
+		_, failures, err := prov.Check(s.new.URN, s.old.Inputs, s.new.Inputs, preview, s.new.SequenceNumber)
+		if err != nil {
+			return rst, nil, err
+		}
+
+		if issueCheckFailures(s.deployment.Diag().Warningf, s.new, s.new.URN, failures) {
+			ref, err := providers.ParseReference(s.Provider())
+			contract.Assert(err == nil)
+
+			pkgName := ref.URN().Type().Name()
+			s.deployment.Diag().Warningf(diag.Message(s.new.URN,
+				"One or more imported inputs failed to validate. "+
```

Comment:
Do you have a full example of what a user sees in this case? Are they able to identify which "one or more inputs" are problematic?

Reply:
```
Diagnostics:
  aws:s3:Bucket (bucket-test):
    warning: aws:s3/bucket:Bucket resource 'bucket-test' has a problem: expected acceleration_status to be one of [Enabled Suspended], got . Examine values at 'Bucket.AccelerationStatus'.
    warning: One or more imported inputs failed to validate. This is almost certainly a bug in the `aws` provider. The import will still proceed, but you will need to edit the generated code after copying it into your program
```